### ISSUE TITLE
plugin stuck "creating context"

### ISSUE THREAD

**Issue by saada** [loft-sh/vcluster-sdk#25]
vcluster: `0.9.1`
vcluster-sdk: `0.1.1`

I based my plugin off of the CRD plugin and tried spinning it up with minimal configurations and I get stuck at this log

```
syncer I0610 21:23:54.685176       1 plugin.go:185] Waiting for plugin crd-sync to register...
crd-sync starting up plugin
crd-sync I0610 21:23:50.688753       1 logr.go:249] plugin: Try creating context...
```

Any suggestions on how to debug this?

**Comment by carlmontanari** [comment]
Hey @saada -- I ran into this spinning up the vcluster-plugin-example the other day as well. Bumping the vcluster-sdk pin -> `0.2.0-beta1` (and tidy to update vendored things) sorted this for me. 

It looks  the proto definition for the plugin registration process changed in the v0.9.x releases of vcluster. So the plugin never registered and we just blocked waiting for that. I suppose you could also use vcluster 0.8.x as an alternative to updating the sdk pin, but It also looks like the cert-manager plugin is doing the same (using 0.2.0-beta1) so I think this is the right thing to do :)

**Comment by FabianKramm** [comment]
@saada @carlmontanari yes you are correct, we had to change the prototype definition for the newer version to allow plugin hooks, unfortunately this requires you to use pre v0.9 vcluster versions for older versions of this SDK. We have now released a stable v0.2.0 version that should work correctly with newer vcluster versions as well.

**Comment by saada** [comment]
Updating worked. Thanks everyone ❤️
